src/App.js

```diff
@@ -1,26 +1,48 @@
 import React, { Component } from "react";
 import FileInput from "./FileInput";
 import Viewer from "./Viewer";
+import _ from "lodash";
 
 export default class App extends Component {
   constructor(props) {
     super(props);
-    this.fr = new FileReader();
-    this.fr.addEventListener("loadend", () => {
-      const file = this.fr.result
-        .split("\n")
-        .map(row => row.split(",")[1])
-        .slice(1)
-        .filter(x => !!x);
-      this.setState({ file });
-    });
-    this.state = { file: null };
+    
+    this.state = { 
+      files: null,
+      types: []
+    };
```

Comment:
is it necessary to maintain a separate array of types? if i'm not mistaken `types === Object.keys(files)`